pkg/printers/humanreadable.go

```diff
@@ -691,12 +724,17 @@ func (h *HumanReadablePrinter) legacyPrinterToTable(obj runtime.Object, handler
 }
 
 // TODO: this method assumes the meta/v1 server API, so should be refactored out of this package
-func printUnstructured(unstructured runtime.Unstructured, w io.Writer, additionalFields []string, options PrintOptions) error {
+func printUnstructured(eventType string, unstructured runtime.Unstructured, w io.Writer, additionalFields []string, options PrintOptions) error {
```

Comment:
I still feel like there is a more elegant construction.  I.e. everything goes through a table for this reason - maybe there should be a Table adapter that wraps table output and unwraps events.  The long term goal was to completely remove everything except a) things that create tables, b) things that output tables

Reply:
I puzzled over this a bit, and without changing the signature of PrintObj, I'm not sure how to accomplish this... the table unwrapping would lose the event type info

The included unit test coverage exercises every output format we have, with and without tables for human-readable output. It's fiddly, but I can live with it for now to get this capability.